projects/admin/src/app/record/detail-view/document-detail-view/files-collections/upload-files/upload-files.component.ts

```diff
@@ -0,0 +1,315 @@
+      return this.translateService.stream('{{ total }} results', { total: remoteTotal });
+    }
+    return totalFiltered === 0
+      ? this.translateService.stream('no result')
+      : this.translateService.stream('{{ total }} results of {{ remoteTotal }}', {
+          total: totalFiltered,
+          remoteTotal: remoteTotal,
+        });
+  }
+
+  /**
+   * Fired when the text to filter the items changes.
+   *
+   * @param event the standard event.
+   */
+  onTextChange($event): void {
+    if (this.filterText.length > 0) {
+      this.filteredFiles = this.files.filter((value) => value.label.toLowerCase().includes(this.filterText.toLowerCase()));
+    } else {
+      this.filteredFiles = this.files;
+    }
+  }
+
+  /**
+   *
+   * @param event the standard event.
+   * @param _ unused.
+   */
+  uploadHandler(event, _) {
+    var obs: Observable<any>;
+    if (event.files.length > 0) {
+      this.spinner.show('file-upload');
+      if (this.parentRecord == null) {
+        // create the parent record
+        // should not happens when a document is used as parent record
+        obs = this.fileService.createParentRecord(this.pid).pipe(
+          map((record) => (this.parentRecord = record)),
+          switchMap(() => {
+            return this.generateCreateRequests(event);
+          })
+        );
+      } else {
+        // the parent record already exists create only the new file
+        obs = this.generateCreateRequests(event);
+      }
+      obs
+        .pipe(
+          catchError((e: any) => {
+            let msg = this.translateService.instant('Server error');
+            if (e.error.message) {
+              msg = `${msg}: ${e.error.message}`;
+            }
+            this.toastrService.error(msg);
+            return of([]);
+          }),
+          tap(() => {
+            this.resetFilter();
+            this.fileUpload.clear();
+            this.toastrService.success(this.translateService.instant('File uploaded successfully.'));
+            this.nUploadedFiles = 0;
+          }),
+        )
+        .subscribe(() => this.spinner.hide('file-upload'));
+    }
+  }
+
+  /**
+   * Generate the sequential http requests.
+   *
+   * @param event the standard event.
+   * @returns an observable of sequential http requests
+   */
+  private generateCreateRequests(event): Observable<any> {
+    return from(event.files).pipe(
+      concatMap((f: any) => this.fileService.create(this.parentRecord.id, f.name, f)),
+      map((file: any) => {
+        this.nUploadedFiles += 1;
+        this.files = [{ label: file.metadata.label ? file.metadata.label : file.key, ...file}, ...this.files];
+        this.filteredFiles =  this.files;
+      }),
+      // like a forkJoin
+      toArray()
+      );
+  }
+
+  /**
+   * Filter the uploaded files.
+   *
+   * @param event the standard event.
+   * @param _ unused.
+   */
+  onSelect(event, _) {
+    const existingFileNames = [];
+    for (var i = 0; i < event.files.length; i++) {
```

Comment:
`let` ?

or use `.map` on event.files  ?

Reply:
unfortunately not it is not a standard array.